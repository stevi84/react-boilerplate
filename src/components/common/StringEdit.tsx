import { TextField } from '@mui/material';
import { useField } from 'formik';

interface StringEditProps {
  id: string;
  name: string;
  label: string;
}

export const StringEdit = (props: StringEditProps) => {
  const { id, name, label } = props;

  const [field, meta] = useField(name);

  return (
    <TextField
      id={id}
      value={field.value}
      onChange={field.onChange}
      label={label}
      error={!!meta.error}
      helperText={meta.error}
      sx={{ margin: 1 }}
    />
  );
};
