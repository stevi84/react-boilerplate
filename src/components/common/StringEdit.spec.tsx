import { describe, it, expect, vi } from 'vitest';
import { StringEdit } from './StringEdit';
import { fireEvent, render, screen } from '@testing-library/react';

const { onChangeMock } = vi.hoisted(() => ({ onChangeMock: vi.fn() }));
vi.mock('formik', () => ({ useField: () => [{ onChange: onChangeMock, value: 'value' }, { error: 'error' }] }));

describe('StringEdit', () => {
  it('should equal saved snapshot', () => {
    const tree = render(<StringEdit id="id" name="name" label="label" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it('should trigger onChange', () => {
    render(<StringEdit id="id" name="name" label="label" />);
    expect(onChangeMock.mock.calls.length).toEqual(0);
    fireEvent.change(screen.getByLabelText('label'), { target: { value: 'test' } });
    expect(onChangeMock.mock.calls.length).toEqual(1);
  });
});
