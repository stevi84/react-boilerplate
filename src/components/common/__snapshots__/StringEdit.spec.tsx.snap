// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StringEdit > should equal saved snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-kn4dux-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="id"
      id="id-label"
    >
      label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-describedby="id-helper-text"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
        id="id"
        type="text"
        value="value"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-kg5swy"
        >
          <span>
            label
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-wq8qsl-MuiFormHelperText-root"
      id="id-helper-text"
    >
      error
    </p>
  </div>
</DocumentFragment>
`;
